src/poprox_recommender/evaluation/evaluate.py

```diff
@@ -0,0 +1,116 @@
+import csv
+import gzip
+import json
+import logging
+
+import numpy as np
+import pandas as pd
+from lenskit.metrics import topn
+from tqdm import tqdm
+
+from poprox_concepts.api.recommendations import RecommendationRequest
+from poprox_recommender.data.mind import MindData
+from poprox_recommender.evaluation.metrics import rank_biased_overlap
+from poprox_recommender.paths import project_root
+
+logger = logging.getLogger("poprox_recommender.measurement_offline")
+
+
+def compute_rec_metric(recs_df: pd.DataFrame, request: RecommendationRequest):
+    # Make sure you run generate.py first to get the recommendations data file
+    recs = recs_df[recs_df["user"] == request.interest_profile.profile_id]
+    truth = mind_data.user_truth(request.interest_profile.profile_id)
```

Comment:
do we remember how this function works and how slow it is?

Reply:
The mind_data.user_truth() works like this:

```
    def user_truth(self, user: UUID) -> pd.DataFrame | None:
        """
        Look up the ground-truth data for a particular user, in LensKit format,
        with item UUIDs for item IDs.
        """
        try:
            uid = self.behavior_id_for_uuid(user)
            imp_log = str(self.behavior_df.loc[uid, "impressions"])
        except KeyError:
            raise ValueError(f"unknown user {user}")

        truth = pd.DataFrame.from_records(
            ((article.split("-")[0], int(article.split("-")[1])) for article in imp_log.split()),
            columns=["mind_item_id", "rating"],
        )
        truth["item"] = [self.news_uuid_for_id(aid) for aid in truth["mind_item_id"]]
        return truth.set_index("item")
```

I remember we discussed during our split of the generation code that this piece can be saved in the evaluate; maybe we can try to run and store truth data in a separate file as well?

Reply:
That's probably not our bottleneck then.

Reply:
It's using the index so should be reasonably fast.